src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/OrderProcessor.kt

```diff
@@ -209,27 +209,33 @@ internal class OrderProcessor(parser: ParserProtocol) : BaseProcessor(parser) {
                 // the v4_parent_subaccount message has subaccountNumber available but v4_orders does not
                 modified.safeSet("marginMode", if (this >= NUM_PARENT_SUBACCOUNTS) MarginMode.isolated.rawValue else MarginMode.cross.rawValue)
             }
-            val size = parser.asDouble(payload["size"])
-            if (size != null) {
-                var totalFilled = parser.asDouble(payload["totalFilled"])
-                var remainingSize = parser.asDouble(payload["remainingSize"])
-                if (totalFilled != null && remainingSize == null) {
-                    remainingSize = size - totalFilled
-                } else if (totalFilled == null && remainingSize != null) {
-                    totalFilled = size - remainingSize
-                }
-                if (totalFilled != null && remainingSize != null) {
+            parser.asDouble(payload["size"])?.let { size ->
+                parser.asDouble(payload["totalFilled"])?.let { totalFilled ->
+                    val remainingSize = size - totalFilled
                     modified.safeSet("totalFilled", totalFilled)
                     modified.safeSet("remainingSize", remainingSize)
-
-                    if (totalFilled != Numeric.double.ZERO && modified["status"] == "OPEN") {
-                        modified.safeSet("status", "PARTIALLY_FILLED")
+                    if (totalFilled != Numeric.double.ZERO && remainingSize != Numeric.double.ZERO) {
+                        when (modified["status"]) {
+                            "OPEN" -> modified.safeSet("status", "PARTIALLY_FILLED")
+                            "CANCELED" -> modified.safeSet("status", "PARTIALLY_CANCELED") // finalized state
+                        }
                     }
                 }
             }
 
             modified.safeSet("cancelReason", payload["removalReason"] ?: payload["cancelReason"])
 
+            parser.asDouble(payload["orderFlags"])?.let { orderFlags ->
+                // if order is short-term order and indexer returns best effort canceled and has no partial fill
+                // treat as a pending order until it's partially filled or finalized
+                val isBestEffortCanceled = modified["status"] == "BEST_EFFORT_CANCELED"
+                val cancelReason = parser.asString(modified["cancelReason"])
+                val isUserCanceled = cancelReason == "USER_CANCELED" || cancelReason == "ORDER_REMOVAL_REASON_USER_CANCELED"
+                if (orderFlags.equals(Numeric.double.ZERO) && isBestEffortCanceled && !isUserCanceled) {
```

Comment:
nit: could we add an extra line of `isShortTermOrder = orderFlags.equals(Numeric.double.ZERO)` just to make it easier to understand